### PR TITLE
Add extensions

### PR DESCRIPTION
**Pull request by gadomski**
**Description:**
Add the usual extension set.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Examples have been updated to reflect changes, if applicable

### CODE REVIEW

File: examples/ocean-heat-content/collection.json [hunk 1]
```diff
@@ -10,6 +10,16 @@
       "type": "application/json",
       "title": "Global Ocean Heat Content CDR"
     },
+    {
+      "rel": "license",
+      "href": "https://www.ncei.noaa.gov/pub/data/sds/cdr/CDRs/Ocean_Heat_Content/UseAgreement_01B-41.pdf",
+      "type": "application/pdf",
+      "title": "CDR Open Data Blurb"
```
Comment (pjhartzell) [inline examples/ocean-heat-content/collection.json:17]: Blurb != scientific language.  🙂

File: examples/ocean-heat-content/collection.json [hunk 2]
```diff
@@ -66,7 +76,56 @@
       "type": "application/json"
     }
   ],
-  "stac_extensions": [],
+  "stac_extensions": [
+    "https://stac-extensions.github.io/scientific/v1.0.0/schema.json",
+    "https://stac-extensions.github.io/item-assets/v1.0.0/schema.json"
+  ],
+  "sci:doi": "10.7289/v53f4mvp",
+  "sci:citation": "Levitus, Sydney; Antonov, John I.; Boyer, Tim P.; Baranova, Olga K.; Garc\u00eda, Hern\u00e1n E.; Locarnini, Ricardo A.; Mishonov, Alexey V.; Reagan, James R.; [Seidov, Dan; Yarosh, Evgeney; Zweng, Melissa M. (2017). NCEI ocean heat content, temperature anomalies, salinity anomalies, thermosteric sea level anomalies, halosteric sea level anomalies, and total steric sea level anomalies from 1955 to present calculated from in situ oceanographic subsurface profile data (NCEI Accession 0164586). [indicate subset used]. NOAA National Centers for Environmental Information. Dataset. https://doi.org/10.7289/v53f4mvp. Accessed [date].",
```
Comment (pjhartzell) [inline examples/ocean-heat-content/collection.json:84]: There are some escaped special characters in here. Can't remember if that is OK or not.

File: examples/ocean-heat-content/collection.json [hunk 3]
```diff
@@ -66,7 +76,56 @@
...
+  "item_assets": {
```
Comment (pjhartzell) [inline examples/ocean-heat-content/collection.json:85]: Any harm in including `raster:bands` here in the `item_assets` objects? They should be the same for each item.

File: examples/ocean-heat-content/ocean-heat-content-1972-03-2000m/ocean-heat-content-1972-03-2000m.json [hunk 4]
```diff
@@ -59,6 +73,15 @@
       "href": "./heat_content_anomaly_0-2000_monthly_1972-03.tif",
       "type": "image/tiff; application=geotiff; profile=cloud-optimized",
       "title": "Ocean Heat Content anomalies from WOA09 : heat_content_anomaly 0-2000 m monthly 1.00 degree 1972-03",
+      "created": "2022-04-18T00:00:00Z",
+      "updated": "2022-04-18T00:00:00Z",
```
Comment (pjhartzell) [inline examples/ocean-heat-content/ocean-heat-content-1972-03-2000m/ocean-heat-content-1972-03-2000m.json:77]: Do we intend to track update datetimes? Is this part of the versioning?

Based on how the NetCDF data is updated, will these always be identical?

File: examples/ocean-heat-content/ocean-heat-content-1972-03-2000m/ocean-heat-content-1972-03-2000m.json [hunk 5]
```diff
@@ -59,6 +73,15 @@
       "href": "./heat_content_anomaly_0-2000_monthly_1972-03.tif",
       "type": "image/tiff; application=geotiff; profile=cloud-optimized",
       "title": "Ocean Heat Content anomalies from WOA09 : heat_content_anomaly 0-2000 m monthly 1.00 degree 1972-03",
```
Comment (pjhartzell) [inline examples/ocean-heat-content/ocean-heat-content-1972-03-2000m/ocean-heat-content-1972-03-2000m.json:75]: Might be worth trying to make the titles more readable/understandable.

**Review by pjhartzell — COMMENTED** [review]
Looking good. I like your approach to cog creation and item creation. Left a few comments.

File: examples/ocean-heat-content/ocean-heat-content-1972-03-2000m/ocean-heat-content-1972-03-2000m.json (hunk 5 above)
Comment (gadomski) [reply examples/ocean-heat-content/ocean-heat-content-1972-03-2000m/ocean-heat-content-1972-03-2000m.json]: I pull this title directly from the NetCDF, so this falls into the "how much extra processing do" categories of issues. I've updated to be _a bit_ clearer, LMK what you think.

File: examples/ocean-heat-content/collection.json (hunk 3 above)
Comment (gadomski) [reply examples/ocean-heat-content/collection.json:85]: No harm, done.

File: examples/ocean-heat-content/collection.json (hunk 1 above)
Comment (pjhartzell) [reply examples/ocean-heat-content/collection.json]: Yikes! Who's letting the kids do science?

File: examples/ocean-heat-content/ocean-heat-content-1972-03-2000m/ocean-heat-content-1972-03-2000m.json (hunk 5 above)
Comment (pjhartzell) [reply examples/ocean-heat-content/ocean-heat-content-1972-03-2000m/ocean-heat-content-1972-03-2000m.json]: I take your point. I'm guessing it's not an easy fix with so many titles. I'm not going to lose sleep over it.

**Review by pjhartzell — APPROVED** [review]

**Pull request merged**